back/engines/commercial/verification/app/controllers/verification/patches/omniauth_callback_controller.rb

```diff
@@ -32,8 +32,12 @@ def verification_callback(verification_method)
               redirect_to url
             rescue VerificationService::VerificationTakenError
               fail_verification('taken')
-            rescue VerificationService::NotEntitledError
-              fail_verification('not_entitled')
+            rescue VerificationService::NotEntitledError => e
+              if e.message == 'not_entitled_under_minimum_age'
+                fail_verification(e.message)
+              else
+                fail_verification('not_entitled')
+              end
             end
```

Comment:
@alexander-cit This satisfies your wish to not use the `why` component, and to not change the behaviour for other errors, but seems rather verbose.

Can you think of any better way to achieve the same thing?

Reply:
It seems I was wrong. It turns out that these errors https://github.com/CitizenLabDotCo/citizenlab/blob/4279dc5e4dbff11b841f8840d8581622afbc9bd6/back/engines/commercial/id_gent_rrn/app/lib/id_gent_rrn/gent_rrn_verification.rb#L74-L75
are not caught here in this file. They are only caught in this action https://github.com/CitizenLabDotCo/citizenlab/blob/7634bef7bfb5e5db0a05a117c84b2d0e8d4ffb47/back/engines/commercial/verification/app/controllers/verification/web_api/v1/verifications_controller.rb#L24-L26

So, it looks like the `rescue VerificationService::NotEntitledError` in this file was not used at all. It means we can do whatever we want with it. Including your previous implementation with `message = e.why ? "not_entitled_#{e.why}" : 'not_entitled'`.

Do you know how to test Oostende RRN? I've never done it before, and it seems you added it.

Reply:
OK, I'll revert to my original approach.

> Do you know how to test Oostende RRN? I've never done it before, and it seems you added it.

I can't remember, and am surprised I seem to have forgotten to add docs (I was sure I did, but can't find them).
There's a lot of info in the ticket and associated PRs, but maybe not sufficient.

If I can't find the docs, I'll try to work it out and write some docs.

Reply:
@alexander-cit I created 2 docs about the Oostende RRN verification method:

[SSO & ID verification / Oostende (BE) RRN [1 customer]](https://www.notion.so/citizenlab/SSO-ID-Verification-263b705ca14e472c8e41b444663e373a?pvs=4#4077f25a1dc94654bb5389b99ece355c)
[SSO & ID: set up locally and test / Oostende RRN](https://www.notion.so/citizenlab/SSO-ID-set-up-locally-and-test-a176dd3e335f47c28f949e5962d32a26?pvs=4#2b76a0390dd642918729ceaccc09c762)